### PR TITLE
fix: Account Resolver

### PR DESCRIPTION
**Pull request by fboucquez**
fix: ConfigLoader split -> MigrationService, AddressesService, ConfigurationUtils
fix: addresses.sinkAddress is not generated when it's not required

Note: I had to fix some imports. If not, oclif errors when generating the docs and the cli metadata. I think it's a circular dependency somewhere.

### CODE REVIEW

File: src/service/AnnounceService.ts [hunk 1]
```diff
@@ -225,16 +226,13 @@ export class AnnounceService {
...
-                        nodeAccount.name,
-                        'signing a transaction',
-                    ),
+                return this.accountResolver.resolveAccount(
                     networkType,
+                    nodeAccount.main,
+                    KeyName.Main,
+                    nodeAccount.name,
+                    'signing a transaction',
+                    'Should not generate!',
```
Comment (gimre-xymcity) [inline src/service/AnnounceService.ts:235]: that error does not seem very descriptive

File: src/service/BootstrapAccountResolver.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        operationDescription: string,
+        generateErrorMessage: string | undefined,
+    ): Promise<Account> {
+        if (!account) {
+            if (generateErrorMessage) {
+                throw new KnownError(generateErrorMessage);
+            }
+            this.logger.info(`Generating ${keyName} account...`);
+            return Account.generateNewAccount(networkType);
+        }
```
Comment (gimre-xymcity) [inline src/service/BootstrapAccountResolver.ts:40]: this seems like exact part from DefaultAccountResolver, refactor?

File: src/service/BootstrapAccountResolver.ts [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                } else {
+                    const enteredAccount = Account.createFromPrivateKey(privateKey, networkType);
+                    if (enteredAccount.publicKey.toUpperCase() !== account.publicKey.toUpperCase()) {
+                        this.logger.info(
+                            `Invalid private key. Expected address is ${address} but you provided the private key for address ${enteredAccount.address.plain()}.\n`,
+                        );
+                        this.logger.info(`Please re-enter private key.`);
+                    } else {
+                        account.privateKey = privateKey;
+                        return Account.createFromPrivateKey(privateKey, networkType);
```
Comment (gimre-xymcity) [inline src/service/BootstrapAccountResolver.ts:69]: return enteredAccount (?)

File: src/service/BootstrapAccountResolver.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                    this.logger.info('Please provide the private key.');
+                } else {
+                    const enteredAccount = Account.createFromPrivateKey(privateKey, networkType);
+                    if (enteredAccount.publicKey.toUpperCase() !== account.publicKey.toUpperCase()) {
+                        this.logger.info(
+                            `Invalid private key. Expected address is ${address} but you provided the private key for address ${enteredAccount.address.plain()}.\n`,
+                        );
+                        this.logger.info(`Please re-enter private key.`);
+                    } else {
+                        account.privateKey = privateKey;
```
Comment (gimre-xymcity) [inline src/service/BootstrapAccountResolver.ts:68]: is setting this deliberate?

File: src/service/AccountResolver.ts [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+ * Basic no prompt implementation. If the account cannot be resolved, it won't be prompted.
+ */
+export class DefaultAccountResolver implements AccountResolver {
+    async resolveAccount(
+        networkType: NetworkType,
+        account: CertificatePair | undefined,
+        keyName: KeyName,
+        nodeName: string,
+        operationDescription: string,
+        generateErrorMessage: string | undefined,
```
Comment (gimre-xymcity) [inline src/service/AccountResolver.ts:45]: is there actually a case (besides tests), where you are NOT providing error message?

(same goes for BootstrapAccountResolver, as theyy implement same interface)

**Review by gimre-xymcity — COMMENTED** [review]
part 1, more tomorrow

**Comment by fboucquez** [comment]
> part 1, more tomorrow

Thanks gimre for the review. I'm trying to allow libs to provide their own accounts cached somewhere else, on-demand, when bootstrap requires it. I'm using this feature in symbol-network, when the keys are pre-created to generate the nemesis block. The feature is working, but the implementation is not great yet.

File: src/service/BootstrapAccountResolver.ts (hunk 4 above)
Comment (fboucquez) [reply src/service/BootstrapAccountResolver.ts:68]: It's to avoid re-prompting for the private key. It's a bit of a secondary effect but it helps.

**Pull request merged**